management_api_app/models/domain/resource.py

```diff
@@ -40,3 +40,12 @@ class Resource(AzureTREModel):
     resourceTemplateParameters: dict = Field({}, title="Resource template parameters", description="Parameters for the deployment")
     deployment: Deployment = Field(Deployment(status=Status.NotDeployed, message=""), title="Deployment", description="Fields related to deployment of this resource")
     deleted: bool = Field(False, title="Is deleted", description="Marks the resource request as deleted (NOTE: this is not the deployment status)")
+
+    def get_resource_request_message_payload(self) -> dict:
```

Comment:
why is this moved here instead of in the service bus code? This code already exists there -- or is it because it is different depending on type? 

It would be great if we can keep this in the service bus class - otherwise we will need to duplicate code for this for other actions etc. (I am working with this specific thing in delete -- see the branch I shared with you earlier) - it doesn't feel like the repositories responsibility.